mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala

```diff
@@ -971,8 +971,10 @@ private class LogisticAggregator(
           val margin = - {
             var sum = 0.0
             features.foreachActive { (index, value) =>
-              if (featuresStd(index) != 0.0 && value != 0.0) {
+              if (featuresStd(index) != 0.0) {
```

Comment:
The previous version is correct. Checking `value != 0.0` is much cheaper than computing `localCoefficientsArray(index)`.